front-end/src/app/grid/grid.component.ts

```diff
@@ -0,0 +1,42 @@
+import { Component, Input, ViewChild } from '@angular/core';
+import { AgGridAngular } from 'ag-grid-angular';
+import { CellClickedEvent, ColDef, GridReadyEvent } from 'ag-grid-community';
+import { Observable, of } from 'rxjs';
+import { LinkRendererComponent } from '../cell-renderers/link-cell/link-cell.component';
+import { GridColumn } from '../types/GridColumn';
+
+@Component({
+  selector: 'app-grid',
+  templateUrl: './grid.component.html',
+  styleUrls: ['./grid.component.css']
+})
+export class GridComponent<T> {
+  @Input() columns: GridColumn[] = [];
+  @Input() rowData$: Observable<T[]> = of();
+  @ViewChild(AgGridAngular) agGrid!: AgGridAngular;
+
+  constructor() {}
+
+  columnDefs: ColDef[] = [];
+
+  defaultColDef: ColDef = {
+    sortable: true,
+    filter: true,
+  };
+
+  onGridReady(params: GridReadyEvent) {
+    this.columnDefs = this.columns.map(({ field, routerLink }) => ({
+      field,
+      cellRenderer: routerLink ? LinkRendererComponent : null,
```

Comment:
This works absolutely fine for the moment - in a bigger project it might be safer to pass in the cellRenderer on the column to avoid complex logic as we could have lots of different custom cell renderers

Reply:
You raise a good point; this may become less maintainable as we start to use more of ag-grid's functionality. The reason I chose this way is because cell renderers are an ag-grid specific concept (with their pros and cons) which I wanted to hide away. Let's review as we carry on...